tests/sentry/sentry_metrics/test_last_seen_updater.py

```diff
@@ -0,0 +1,79 @@
+from datetime import datetime
+from unittest.mock import Mock
+
+import pytest
+from arroyo import Message, Partition, Topic
+from arroyo.backends.kafka import KafkaPayload
+
+from sentry.metrics.dummy import DummyMetricsBackend
+from sentry.sentry_metrics.last_seen_updater import (
+    LastSeenUpdaterMessageFilter,
+    retrieve_db_read_keys,
+)
+from sentry.testutils.helpers import override_options
+
+
+def test_retrieve_db_read_keys_meta_field_present_with_db_keys():
```

Comment:
I think you can test this much more deeply.
Arroyo strategies are designed to not depend directly on an instance of a Consumer (they can have other dependencies that make testing hard but it is not the case here).

I would test it this way: the filter is well tested. 
I would similarly test the collector. You can update postgres in tests. See the indexer tests.

Then you can run a couple of tests for your entire strategy.
You can instantiate it by creating the factory with this.
https://github.com/getsentry/sentry/pull/34865/files#diff-8187f0eac40e1c087ce1371f184b76039739f4a0abc32e7fc70dc0c98efc9693R117-R126

The factory has a create method, you can just call it and get the instance of the strategy.
Then you can submit messages, call poll, and after a few messages call join, so you can see everything works together.

Reply:
I think I've added the testing you're looking for.